### PR TITLE
Pull request to implement issue 298

### PR DESCRIPTION
**Pull request by jamgold**
I modified the `_getUsersInRoleCursor` method to recognize `onlyScoped` option as well and added a server side test

### CODE REVIEW

File: roles/tests/server.js [hunk 1]
```diff
@@ -38,6 +38,11 @@ describe('roles', function () {
     var userId = users[username]
     var userObj = Meteor.users.findOne({ _id: userId })
 
+    // console.log(`testUser server scope=${typeof scope}`);
```
Comment (SimonSimCity) [inline roles/tests/server.js]: Please remove this line

**Comment by SimonSimCity** [comment]
I'll do a more thorough look at this after the weekend, but as of now it already looks good.

File: roles/tests/server.js (hunk 1 above)
Comment (jamgold) [reply roles/tests/server.js]: Done

File: roles/roles_common.js [hunk 2]
```diff
@@ -780,7 +780,13 @@ Object.assign(Roles, {
       return roles
     }
 
-    return [...new Set(roles.map(r => r.inheritedRoles || [r.role]).reduce((rev, current) => rev.concat(current), []).map(r => r._id))]
+    return [
+      ...new Set(roles
+        .map(r => r.inheritedRoles || [r.role])
+        .reduce((rev, current) => rev.concat(current), [])
+        .map(r => r ? r._id : null)
```
Comment (SimonSimCity) [inline roles/roles_common.js]: In which case would `r` be `null`? And how is this related to this PR?

File: roles/tests/client.js [hunk 3]
```diff
@@ -28,19 +28,22 @@ describe('roles', function () {
     }
   }
 
-  function testUser (username, expectedRoles, group) {
+  function testUser (username, expectedRoles, scope) {
     var user = users[username]
-
+    // console.log(`testUser client scope=${typeof scope}`);
+    if(typeof scope == 'string') {
```
Comment (SimonSimCity) [inline roles/tests/client.js]: Is there a reason for changing this ..? It should work either way, shouldn't it? Now it seems like we test one case less ...

File: roles/tests/server.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ describe('roles', function () {
       return
     }
 
-    return Meteor.roleAssignment.find({ _id: loggedInUserId })
+    return Meteor.roleAssignment.find( { _id: loggedInUserId });
```
Comment (SimonSimCity) [inline roles/tests/server.js]: At least the added semicolon should not pass the check for code-formatting. Please also evaluate it for the rest of the changes.

File: roles/tests/server.js [hunk 5]
```diff
@@ -38,6 +38,10 @@ describe('roles', function () {
     var userId = users[username]
     var userObj = Meteor.users.findOne({ _id: userId })
 
+    if (typeof scope == 'string') {
```
Comment (SimonSimCity) [inline roles/tests/server.js]: Is this change needed? I guess it's the same as in the client test file.
